dashboard/dashboard.py

```diff
@@ -0,0 +1,777 @@
+    Retrieves a list of tags from the database for a specific case
+    """
+    query = """
+            SELECT t.tag_name 
+            FROM tag_assignment as ta
+            JOIN tag as t ON t.tag_id = ta.tag_id
+            JOIN court_case as cc ON cc.court_case_id = ta.court_case_id
+            WHERE cc.title = %s;
+    """
+    with cnx.cursor(cursor_factory=RealDictCursor) as curs:
+        curs.execute(query, (title,))
+        result = curs.fetchall()
+    return [row["tag_name"] for row in result]
+
+
+def get_participants_and_lawyers_for_case(cnx: connection, title: str,
+                                          is_defendant: bool) -> list[dict]:
+    """
+    Retrieves a list of prosecutors from the database for a specific case
+    """
+    query = """
+            SELECT p.participant_name, l.lawyer_name, lf.law_firm_name, cc.title
+            FROM participant_assignment as pa
+            JOIN participant as p ON p.participant_id = pa.participant_id
+            JOIN court_case as cc ON cc.court_case_id = pa.court_case_id
+            JOIN lawyer as l ON l.lawyer_id = pa.lawyer_id
+            JOIN law_firm as lf ON lf.law_firm_id = l.law_firm_id
+            WHERE cc.title = %s 
+            AND pa.is_defendant = %s;
+    """
+    with cnx.cursor(cursor_factory=RealDictCursor) as curs:
+        curs.execute(query, (title, is_defendant))
+        result = curs.fetchall()
+    return result
+
+
+def format_participants_to_string(participants: list[dict]) -> str:
+    """
+    Get all participants and format them to the desired html to then display
+    """
+    html = ""
+    current_participants = []
+    for row in participants:
+        participant = row.get("participant_name").title()
+        if participant not in current_participants:
+            current_participants.append(participant)
+        law_firm = row.get("law_firm_name")
+        lawyer = row.get("lawyer_name")
+        html += f"<div><strong>{participant}</strong><br>"
+        if lawyer and lawyer != 'None':
+            html += f"Represented by: {lawyer.title()} <br>"
+        if law_firm and law_firm != 'None':
+            html += f"From: {law_firm} <br>"
+        html += '<br>'
+        html += "</div>"
+    return html
+
+
+def display_claimants_and_defendants(selected_case):
+    """
+    Special display for  claimants and defendants so that it stands out
+    """
+    st.divider()
+    container = st.container(border=False)
+    col1, col2, col3 = container.columns([5, 1, 5])
+    with col1:
+        st.markdown("""<span style='color:white'>Claimants</span>""",
+                    unsafe_allow_html=True)
+        prosecuting_participants = get_participants_and_lawyers_for_case(
+            conn, selected_case, False)
+        prosecuting_participants_html = format_participants_to_string(
+            prosecuting_participants)
+        st.markdown(prosecuting_participants_html,
+                    unsafe_allow_html=True)
+    with col2:
+        pass
```

Comment:
What is this for?

Reply:
This is for the formatting of the dashboard, to make the spaces between graphs organised.